clang/test/SemaSYCL/loop_fusion.cpp

```diff
@@ -51,13 +54,34 @@ template <int N>
 [[intel::loop_fuse(N)]] void func17(); // expected-error{{'loop_fuse' attribute requires a non-negative integral compile time constant expression}}
 
 template <typename Ty>
-[[intel::loop_fuse(Ty{})]] void func18() {} // expected-error{{'loop_fuse' attribute requires an integer constant}}
+[[intel::loop_fuse(Ty{})]] void func18() {} // expected-error{{integral constant expression must have integral or unscoped enumeration type, not 'float'}}
+
+template <int size>
+[[intel::loop_fuse(12)]] void func19();     // expected-note {{previous attribute is here}}
+template <int size>
+[[intel::loop_fuse(size)]] void func19() {} // expected-warning {{attribute 'loop_fuse' is already applied with different arguments}}
+
+template <int size>
+[[intel::loop_fuse_independent(5)]] void func20();      // expected-note {{previous attribute is here}}
+template <int size>
+[[intel::loop_fuse_independent(size)]] void func20() {} // expected-warning {{attribute 'loop_fuse_independent' is already applied with different arguments}}
```

Comment:
Test cases cased assertion with existing implementation. This is fixed now

Reply:
Great catch!